LibreNMS/Config.php

```diff
@@ -137,6 +137,41 @@ public static function get($key, $default = null)
         return Arr::get(self::$config, $key, $default);
     }
 
+    public static function getExecutable(string $key): string
+    {
+        $binary = self::get($key, $key);
+
+        if (Str::startsWith($binary, '/')) {
+            return $binary;
+        }
+
+        $output = `whereis -b $binary`;
```

Comment:
What happens when whereis doesn't exist?

Reply:
Is bash guaranteed @mpikzink ?

I know the daily.sh script requires bash, but I"m not sure about any other scripts.

Reply:
`whereis` doesn't only rely on `PATH` to find binaries, I guess thats why we use it.
But if `PATH` is not set, `whereis` itself can't be found :) cat and mouse

Reply:
`whereis whereis` lol

Reply:
Since @laf asked about relevance, I remembered the discussion and read up on it a bit because it interested me. With a POSIX shell, you can achieve similar behaviour with ‘command -v bash’. I have also seen this in some Debian ‘.deb’ packages. With POSIX compatibility, we don't cover all cases either, but it will cover most cases.